openhtf/conf.py

```diff
@@ -294,334 +353,110 @@ def LoadFromDict(self, dictionary, force_reload=False):
-    config = Config()
-    if config.url:
-      print config.url
-  """
-  model = ConfigModel()
-
-  def __init__(self, model=None):
-    """Initializes the configuration object with its shared state.
-
-    Args:
-      model: The data model to use, defaults to the one shared amonst all config
-          objects.
-    """
-    self.model = model or Config.model
-
-  # pylint: disable=missing-docstring
-  @property
-  def dictionary(self):
-    if not self.loaded:
-      raise ConfigurationNotLoadedError()
-    return self.model.state
-
-  @property
-  def loaded(self):
-    return self.model.loaded
-
-  # pylint: enable=missing-docstring
-
-  def __getattr__(self, name):  # pylint: disable=invalid-name
-    """Searches for the value in our config, returning if its found.
-
-    Args:
-      name: name of attribute
-    Returns:
-      None if key not found and is not required, otherwise the value.
-    Raises:
-      MissingRequiredKeyError: If the key was declared required and
-          is not found.
-      UndeclaredKeyAccessError: If the key being accessed was not
-          declared.
-      ConfigurationNotLoadedError: If the config file has not been loaded, this
-          typically means you accessed the config at import time.
-    """
-    if not self.model.loaded:
-      raise ConfigurationNotLoadedError(name)
-    return self.model.GetValue(name)
-
-  def __contains__(self, name):  # pylint: disable=invalid-name
-    """Provides the ability to quickly check if a config key is declared."""
-    return self.model.ContainsKey(name)
-
-  def __getitem__(self, key):  # pylint: disable=invalid-name
-    """Allows access to config items via an indexer."""
-    return self.__getattr__(key)
-
-  def __repr__(self):
-    return '<Config (loaded: %s): 0x%x>' % (self.model.loaded, id(self))
-
-  def CreateStackedConfig(self, model):
-    """Stacks a new model onto the current model, creating a new config.
+  def _asdict(self):
+    """Create a dictionary snapshot of the current config values."""
+    # Start with any default values we have, and override with loaded values,
+    # and then override with flag values.
+    retval = {key: self._declarations[key].default_value for
+              key in self._declarations if self._declarations[key].has_default}
+    retval.update(self._loaded_values)
+    # Only update keys that are declared so we don't allow injecting
+    # un-declared keys via commandline flags.
+    for key, value in self._flag_values:
+      if key in retval:
+        retval[key] = value
+    return retval
+
+  def InjectPositionalArgs(self, method):
+    """Decorator for injecting positional arguments from the configuration.
+
+    This decorator wraps the given method, so that any positional arguments are
+    passed with corresponding values from the configuration.  The name of the
+    positional argument must match the configuration key.  Keyword arguments
```

Comment:
Keyword arguments in the signature or when invoked? Why would keyword arguments that aren't being modified need to match config keys? That feels wrong to me. Also, the implementation seems to do the opposite, where it only warns when you have a kwarg passed in that _is_ in the config.